traveler-swift-core/TravelerKit/Models/OrderResult.swift

```diff
@@ -72,6 +72,34 @@ public struct OrderResult: Decodable {
         return OrderResult(total: result.total, orders: orders, fromDate: result.fromDate, toDate: result.toDate)
     }
 
+    /**
+     Updates `Order` in orders in self.
+
+     - Parameters:
+     - order: The `Order` to be updated
+
+     - Returns: An updated `OrderResult` if the the `Order` exists; nil otherwise.
+     */
+
+    public func update(_ order: Order) -> OrderResult? {
+        var updatedOrders = self.orders
+        var updated = false
+
+        for key in updatedOrders.keys {
+            if order == updatedOrders[key] {
+                updatedOrders[key] = order
+
+                updated = true
+            }
+        }
+
+        guard updated == true else {
+            return nil
+        }
+
+        return OrderResult(total: self.total, orders: updatedOrders, fromDate: self.fromDate, toDate: self.toDate)
+    }
```

Comment:
@ata-n What do you think about this approach?. I think it works better, I was reading that for immutable objects, a way to change state is through the builder pattern. Additionally, If I'm not mistaken, this approach also preserves any previous merging. It preserves immutability yet avoids the need of doing anything fancy with classes (and its fairly understandable for most developers).

Reply:
I dont like that it can return nil...makes code paths difficult.

Also Swift has `mutating` keyword for instances like these, which leaves the option of mutability to the developer

I dont know how I feel about traversing the entire results either...hmmmm..its a thinker